README.md
## Bean Pricing Tables

[Bean Pricing Tables](http://themebeans.com/plugin/bean-pricing-tables/) is a WordPress plugin by [ThemeBeans](http://themebeans.com) that adds a [shortcode](https://codex.wordpress.org/Shortcode) for displaying pricing tables on your posts and pages.

The shortcode can be used to display 1 column, 2 column, or 3 column responsive pricing tables. The plugin installs a shortcode generator on the [TinyMCE](https://codex.wordpress.org/TinyMCE) panel as well.

For more information, check out the [plugin page](http://themebeans.com/plugin/bean-pricing-tables/).


## How to Install

1. Copy the plugin folder 'bean-pricingtables' to the '/wp-content/plugins' directory of your WordPress installation.
2. Go to the _WordPress Dashboard > Plugins_ screen and click _Activate_ under _Bean Pricing Tables_.


## How to Use

1. In the visual editor (TinyMCE editor) of your posts' and pages' edit/add screen, click the table icon in the top bar.
2. Select any of the table styles.
3. Analyze the newly added shortcode on your editor, and customize it according to the needs.


changelog.txt
Apr 14 2014 - Version 1.3
<ul>
<li>WP 3.9 TinyMCE editor fix</li>
</ul>

Feb 14 2014 - Version 1.2.1
<ul>
<li>Removed license input from settings menu</li>
</ul>

Feb 12 2014 - Version 1.2
<ul>
<li>Added support for auto activation</li>
</ul>

Jan 1 2014 - Version 1.1
<ul>
<li>Added a Plugin Updater. Nice!</li>
</ul>

Dec 12 2013 - Version 1.0
<ul>
<li>Initial Release</li>
</ul>

tinymce/plugin.js
// closure to avoid namespace collision
(function () {
    //ADD PRICING TABLE PLUGIN
    tinymce.PluginManager.add("BeanPricingTables", function(editor, url) {
        //CREATE THE BUTTON
        editor.addButton('bean_pricingtable_button', {
            type: "splitbutton",
            title: "Bean Pricing Tables", //BUTTON TITLE
            menu: [
                 createSubmenuButtonImmediate( "Pricing Table, 1 Column",
                    '[pricing_table columns="1"][pricing_column highlight="true"][price_info title="Free" cost="Zero dollars." url="http://themebeans.com" highlighted="Most Popular"]This is a quick description or something of the free package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>32GB</span> of Free Space</li><li>Completely Editable</li><li>Mobile Ready</li></ul>[/pricing_column][/pricing_table]' 
                    ),
                 createSubmenuButtonImmediate( "Pricing Table, 2 Column",
                    '[pricing_table columns="2"][pricing_column][price_info title="Free" cost="Zero dollars." url="http://themebeans.com" highlighted="Most Popular"]This is a quick description or something of the free package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>32GB</span> of Free Space</li><li>Completely Editable</li><li>Mobile Ready</li></ul>[/pricing_column][pricing_column][price_info title="Basic" cost="$5.99 / month." url="http://themebeans.com"]This is a quick description or something of the basic package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>64GB</span> of Free Space</li><li>Completely Editable</li><li>Loads more Features</li></ul>[/pricing_column][/pricing_table]' 
                    ),
                 createSubmenuButtonImmediate( "Pricing Table, 3 Column",
                    '[pricing_table columns="3"][pricing_column][price_info title="Free" cost="Zero dollars." url="http://themebeans.com"]This is a quick description of the free package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>32GB</span> of Free Space</li><li>Completely Editable</li><li>Mobile Ready</li></ul>[/pricing_column][pricing_column][price_info title="Basic" cost="$5.99 / month." url="http://themebeans.com" highlighted="Most Popular"]This is a quick description of the basic package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>64GB</span> of Free Space</li><li>Completely Editable</li><li>Loads more Features</li></ul>[/pricing_column][pricing_column][price_info title="Premium" cost="$19.99 / month" url="http://themebeans.com"]This is a quick description of the premium package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>128GB</span> of Free Space</li><li>Completely Editable</li><li>Loads more Features</li></ul>[/pricing_column][/pricing_table]'
                     )
            ],
            onclick: function() {}
        });

        function createSubmenuButtonImmediate( title, sc ) {
            return {
                text: title,
                onclick: function() {
                    executeTinyMCECommand( 'mceInsertContent', sc );
                }
            }
        }

        function executeTinyMCECommand( command, args ) {
            if (typeof window.tinyMCE.activeEditor != 'undefined') {
                window.tinyMCE.activeEditor.selection.moveToBookmark(window.tinymce_cursor);
            }
            if (typeof window.tinyMCE.execInstanceCommand != 'undefined') {
                window.tinyMCE.execInstanceCommand('content', command, false, args);

            } else {
                if (typeof window.tinyMCE.execCommand != 'undefined') {
                    window.tinyMCE.get('content').execCommand(command, false, args);
                }
            }
        }
	});
})();

tinymce/plugin-w3.9.js
// closure to avoid namespace collision
(function () {
	// create the plugin
	tinymce.create("tinymce.plugins.BeanPricingTables", {

		//CREATES CONTROL INSTANCES BASED ON CONTROL'S ID - OUR BUTTON ID IS bean_pricingtables_button
		createControl: function ( btn, e ) {console.log(btn);
			if ( btn == "bean_pricingtable_button" ) {

				var a = this;

				//CREATE THE BUTTON
				var btn = e.createSplitButton('bean_pricingtable_button', {
                    title: "Bean Pricing Tables",
                });

				//RENDER A DROPDOWN MENU
                btn.onRenderMenu.add(function (c, b) {
					a.addImmediate( b, "Pricing Table, 1 Column",
					'[pricing_table columns="1"][pricing_column highlight="true"][price_info title="Free" cost="Zero dollars." url="http://themebeans.com" highlighted="Most Popular"]This is a quick description or something of the free package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>32GB</span> of Free Space</li><li>Completely Editable</li><li>Mobile Ready</li></ul>[/pricing_column][/pricing_table]'
					);
					a.addImmediate( b, "Pricing Table, 2 Column",
					'[pricing_table columns="2"][pricing_column][price_info title="Free" cost="Zero dollars." url="http://themebeans.com" highlighted="Most Popular"]This is a quick description or something of the free package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>32GB</span> of Free Space</li><li>Completely Editable</li><li>Mobile Ready</li></ul>[/pricing_column][pricing_column][price_info title="Basic" cost="$5.99 / month." url="http://themebeans.com"]This is a quick description or something of the basic package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>64GB</span> of Free Space</li><li>Completely Editable</li><li>Loads more Features</li></ul>[/pricing_column][/pricing_table]'
					);
					a.addImmediate( b, "Pricing Table, 3 Column",
					'[pricing_table columns="3"][pricing_column][price_info title="Free" cost="Zero dollars." url="http://themebeans.com"]This is a quick description of the free package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>32GB</span> of Free Space</li><li>Completely Editable</li><li>Mobile Ready</li></ul>[/pricing_column][pricing_column][price_info title="Basic" cost="$5.99 / month." url="http://themebeans.com" highlighted="Most Popular"]This is a quick description of the basic package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>64GB</span> of Free Space</li><li>Completely Editable</li><li>Loads more Features</li></ul>[/pricing_column][pricing_column][price_info title="Premium" cost="$19.99 / month" url="http://themebeans.com"]This is a quick description of the premium package.[/price_info]<ul><li class="info" title="An incredibly intuitive pricing tables plugin. Generated via shortcodes so its nice and easy.">Bean Pricing Tables</li><li class="info" title="This is an additional info hover. Simply add the info class, and a title to your list item.">List Hover Example</li><li class="info" title="Just surround your text in a span HTML element to highlight elements"><span>128GB</span> of Free Space</li><li>Completely Editable</li><li>Loads more Features</li></ul>[/pricing_column][/pricing_table]'
					);


				});

                return btn;
			}

			return null;
		},

		//INSERT SHORTCODE AS DIRECT CONTENT
		addImmediate: function ( ed, title, sc) {
			ed.add({
				title: title,
				onclick: function () {
					tinyMCE.activeEditor.execCommand( "mceInsertContent", false, sc )
				}
			})
		},

		//CREDS
		getInfo: function () {
			return {
				longname : "ThemeBeans Shortcodes",
				author : 'ThemeBeans',
				authorurl : 'http://themebeans.com/',
				infourl : 'http://themebeans.com/plugin/bean-shortcodes-plugin',
				version : "1.0"
			};
		}
	});

	//ADD PRICING TABLE PLUGIN
	tinymce.PluginManager.add("BeanPricingTables", tinymce.plugins.BeanPricingTables);
})();

updates/EDD_SL_Setup.php
<?php
/**
 * This file sets up the license updater.
 *
 *  
 * @package Bean Plugins
 * @subpackage BeanPricingTables
 * @author ThemeBeans
 * @since BeanPricingTables 1.2
 */
 
/*===================================================================*/
/* REGISTER OPTION
/*===================================================================*/
function edd_beanpricingtables_register_option() 
{
	register_setting('edd_beanpricingtables_license', 'edd_beanpricingtables_license_key', 'edd_bean_pricingtables_sanitize_license' );
}
add_action('admin_init', 'edd_beanpricingtables_register_option');

function edd_bean_pricingtables_sanitize_license( $new ) 
{
	$old = get_option( 'edd_beanpricingtables_license_key' );
	if( $old && $old != $new ) {
		delete_option( 'edd_beanpricingtables_license_status' );
	}
	return $new;
}




/*===================================================================*/
/* ACTIVATE LICENSE KEY
/*===================================================================*/
function edd_beanpricingtables_activate_license() 
{
	//LISTEN FOR ACTIVATE BUTTON
	if( isset( $_POST['edd_beanpricingtables_license_activate'] ) ) {

		//SECUIRTY CHECK
	 	if( ! check_admin_referer( 'edd_beanpricingtables_nonce', 'edd_beanpricingtables_nonce' ) ) 	
			return; //GET OUT IF WE DIDNT CLICK ACTIVATE

		//RETRIEVE LICENSE FROM DATABASE
		$license = trim( get_option( 'edd_beanpricingtables_license_key' ) );
			
		//DATA TO SEND
		$api_params = array( 
			'edd_action'=> 'activate_license', 
			'license' 	=> $license, 
			'item_name' => urlencode( BEANPRICINGTABLES_EDD_TB_NAME )
		);
		
		//CALL CUSTOM API
		$response = wp_remote_get( add_query_arg( $api_params, BEANPRICINGTABLES_EDD_TB_URL ), array( 'timeout' => 15, 'sslverify' => false ) );

		//MAKE SURE RESPONSE IS GOOD
		if ( is_wp_error( $response ) )
			return false;

		//DECODE LICENSE DATA
		$license_data = json_decode( wp_remote_retrieve_body( $response ) );
		update_option( 'edd_beanpricingtables_license_status', $license_data->license );
	}
}
add_action('admin_init', 'edd_beanpricingtables_activate_license');




/*===================================================================*/
/* DEACTIVATE LICENSE KEY
/*===================================================================*/
function edd_beanpricingtables_deactivate_license() 
{
	//LISTEN FOR ACTIVATION CLICK
	if( isset( $_POST['edd_beanpricingtables_license_deactivate'] ) ) 
	{
		//SECURITY CHECK
	 	if( ! check_admin_referer( 'edd_beanpricingtables_nonce', 'edd_beanpricingtables_nonce' ) ) 	
			return; // get out if we didn't click the Activate button

		//RETRIEVE LICENSE FROM DATABASE
		$license = trim( get_option( 'edd_beanpricingtables_license_key' ) );

		//DATA TO SEND
		$api_params = array( 
			'edd_action'=> 'deactivate_license', 
			'license' 	=> $license, 
			'item_name' => urlencode( BEANPRICINGTABLES_EDD_TB_NAME ) // the name of our product in EDD
		);
		
		//CALL CUSTOM API
		$response = wp_remote_get( add_query_arg( $api_params, BEANPRICINGTABLES_EDD_TB_URL ), array( 'timeout' => 15, 'sslverify' => false ) );

		//MAKE SURE RESPONSE IS GOOD
		if ( is_wp_error( $response ) )
			return false;

		//DECODE LICENSE DATA
		$license_data = json_decode( wp_remote_retrieve_body( $response ) );
		
		//DECODE LICENSE DATA
		if( $license_data->license == 'deactivated' )
			delete_option( 'edd_beanpricingtables_license_status' );
	}
}
add_action('admin_init', 'edd_beanpricingtables_deactivate_license');

bean-theme-pricingtables.php
<?php
/**
 * The file controls the shortcode that is added to the Visual Editor.
 *
 *  
 * @package Bean Plugins
 * @subpackage PricingTables
 * @author ThemeBeans
 * @since PricingTables 1.0
 */
 
 
 
 
/*===================================================================*/
/*  TEXT WIDGET SHORTCODE FILTERS
/*===================================================================*/
add_filter('widget_text', 'shortcode_unautop', 10);
add_filter('widget_text', 'do_shortcode', 10);




/*===================================================================*/
/*  PRICING TABLE SHORTCODE
/*===================================================================*/
function bean_pricing_table( $atts, $content = null ) {
	global $shortcode_pricing_table;
	extract(shortcode_atts(array(
		'columns' => '3'
    ), $atts));

	switch ($columns) {
		case '1':
			$columnsClass = 'one-column-table';
			$widthClass = 'five columns mobile-four centered';
			break;
		case '2':
			$columnsClass = 'two-column-table';
			$widthClass = 'six columns mobile-four';
			break;
		case '3':
			$columnsClass = 'three-column-table';
			$widthClass = 'four columns mobile-two';
			break;
		case '4':
			$columnsClass = 'four-column-table';
			break;
	}

	do_shortcode($content);

	$columnContent = '';

	if (is_array($shortcode_pricing_table)) {

		for ($i = 0; $i < count($shortcode_pricing_table); $i++) {


			$colClass = 'pricing-column'; $n = $i + 1;
			$columnContent .= '<div class="'.$colClass.' '.$widthClass.'">';
			$columnContent .= str_replace(array("\r\n", "\n", "\r"), array("", "", ""), $shortcode_pricing_table[$i]['content']); str_replace('', '', $shortcode_pricing_table[$i]['content']);
			$columnContent .= '</div>';
		}

		$finished_table = '<br><div class="bean-pricing-table '.$columnsClass.'">'.$columnContent.'</div>';
	}
	$shortcode_pricing_table = '';

	return $finished_table;

}
add_shortcode('pricing_table', 'bean_pricing_table');




/*===================================================================*/
/*  SINGLE COLUMN
/*===================================================================*/
function shortcode_pricing_column( $atts, $content = null ) {
	global $shortcode_pricing_table;
	extract(shortcode_atts(array(
    ), $atts));

	if ( isset( $highlight ) ) {
		$highlight = strtolower($highlight);
	}
	$column['content'] = do_shortcode($content);
	$shortcode_pricing_table[] = $column;


}
add_shortcode('pricing_column', 'shortcode_pricing_column');




/*===================================================================*/
/*  PRICE INFO
/*===================================================================*/
function bean_shortcode_price_info( $atts, $content = null ) {
	global $shortcode_pricing_table;
	extract(shortcode_atts(array(
		'title' => '',
		'url' => '',
		'highlighted' => '',
		'cost' => ''
    ), $atts));

	$price_info = '';
	
	if ($highlighted) $price_info = '<div class="pricing-highlighted">'.$highlighted.'</div>';
	if ($url) $price_info .= '<a href="'.$url.'">';
		$price_info .= '<div class="table-mast">';
			if ($title) $price_info .= '<h5 class="title">'.$title.'</h5>';
			if ($cost) $price_info .= '<h6 class="price">'. $cost .'</h6>';
			if ($content) $price_info .= '<div class="details"><p>'. do_shortcode($content) .'</p></div>';
		$price_info .= '</div>';
	if ($url) $price_info .= '</a>';

	return $price_info;

}
add_shortcode('price_info', 'bean_shortcode_price_info');
?>

updates/EDD_SL_Activation.php
<?php
/**
 * This file sets up the auto license activater.
 *
 *
 * @package Bean Plugins
 * @subpackage BeanSocial
 * @author ThemeBeans
 * @since BeanSocial 1.4
 */
 
 
/*===================================================================*/
/* ACTIVATION
/*===================================================================*/
function edd_beanpricingtables_activate_license() 
{
	//GET STATUS
	$status = get_option( 'edd_beanpricingtables_license_status' );
	
	//RETRIEVE LICENSE KEY
	$license = trim( get_option( 'edd_beanpricingtables_activate_license' ) );
	
	//UNCOMMENT TO TEST LICENSE INPUT	
	//echo 'License'; echo $license; echo ' / Status'; echo $status;
	
	//CHECK IF STATUS IS INVALID
	if( $status == '' OR $status == 'invalid' ) 
	{
		//DATA TO SEND WITH OUR API REQUEST
		$api_params = array( 
			'edd_action'=> 'activate_license', 
			'license' 	=> $license, 
			'item_name' => urlencode( EDD_BEANPRICINGTABLES_NAME )
		);
		
		//CALL CUSTOM API
		$response = wp_remote_get( add_query_arg( $api_params, EDD_BEANPRICINGTABLES_TB_URL ), array( 'timeout' => 15, 'sslverify' => false ) );
	
		//MAKE SURE RESPONSE IS OK
		if ( is_wp_error( $response ) )
			return false;
	
		//DECODE DATA
		$license_data = json_decode( wp_remote_retrieve_body( $response ) );
		
		//$license_data->license will be either "active" or "inactive"
	
		update_option( 'edd_beanpricingtables_license_status', $license_data->license );
	}
}
add_action('admin_init', 'edd_beanpricingtables_activate_license');

updates/EDD_SL_Plugin_Updater.php
<?php
/**
 * This file controls plugin updater API calls.
 *
 *  
 * @package Bean Plugins
 * @subpackage BeanPricingTables
 * @author ThemeBeans
 * @since BeanPricingTables 1.2
 */
 
//UNCOMMENT THIS LINE FOR TESTING
//set_site_transient( 'update_plugins', null );

/**
 * Allows plugins to use their own update API.
 *
 * @author Pippin Williamson
 * @version 1.1
 */
class EDD_SL_Plugin_Updater {
	private $api_url  = '';
	private $api_data = array();
	private $name     = '';
	private $slug     = '';

	/**
	 * Class constructor.
	 *
	 * @uses plugin_basename()
	 * @uses hook()
	 *
	 * @param string $_api_url The URL pointing to the custom API endpoint.
	 * @param string $_plugin_file Path to the plugin file.
	 * @param array $_api_data Optional data to send with API calls.
	 * @return void
	 */
	function __construct( $_api_url, $_plugin_file, $_api_data = null ) {
		$this->api_url  = trailingslashit( $_api_url );
		$this->api_data = urlencode_deep( $_api_data );
		$this->name     = plugin_basename( $_plugin_file );
		$this->slug     = basename( $_plugin_file, '.php');
		$this->version  = $_api_data['version'];

		// Set up hooks.
		$this->hook();
	}

	/**
	 * Set up Wordpress filters to hook into WP's update process.
	 *
	 * @uses add_filter()
	 *
	 * @return void
	 */
	private function hook() {
		add_filter( 'pre_set_site_transient_update_plugins', array( $this, 'pre_set_site_transient_update_plugins_filter' ) );
		add_filter( 'plugins_api', array( $this, 'plugins_api_filter' ), 10, 3 );
	}

	/**
	 * Check for Updates at the defined API endpoint and modify the update array.
	 *
	 * This function dives into the update api just when Wordpress creates its update array,
	 * then adds a custom API call and injects the custom plugin data retrieved from the API.
	 * It is reassembled from parts of the native Wordpress plugin update code.
	 * See wp-includes/update.php line 121 for the original wp_update_plugins() function.
	 *
	 * @uses api_request()
	 *
	 * @param array $_transient_data Update array build by Wordpress.
	 * @return array Modified update array with custom plugin data.
	 */
	function pre_set_site_transient_update_plugins_filter( $_transient_data ) {


		if( empty( $_transient_data ) ) return $_transient_data;

		$to_send = array( 'slug' => $this->slug );

		$api_response = $this->api_request( 'plugin_latest_version', $to_send );

		if( false !== $api_response && is_object( $api_response ) && isset( $api_response->new_version ) ) {
			if( version_compare( $this->version, $api_response->new_version, '<' ) )
				$_transient_data->response[$this->name] = $api_response;
	}
		return $_transient_data;
	}


	/**
	 * Updates information on the "View version x.x details" page with custom data.
	 *
	 * @uses api_request()
	 *
	 * @param mixed $_data
	 * @param string $_action
	 * @param object $_args
	 * @return object $_data
	 */
	function plugins_api_filter( $_data, $_action = '', $_args = null ) {
		if ( ( $_action != 'plugin_information' ) || !isset( $_args->slug ) || ( $_args->slug != $this->slug ) ) return $_data;

		$to_send = array( 'slug' => $this->slug );

		$api_response = $this->api_request( 'plugin_information', $to_send );
		if ( false !== $api_response ) $_data = $api_response;

		return $_data;
	}

	/**
	 * Calls the API and, if successfull, returns the object delivered by the API.
	 *
	 * @uses get_bloginfo()
	 * @uses wp_remote_post()
	 * @uses is_wp_error()
	 *
	 * @param string $_action The requested action.
	 * @param array $_data Parameters for the API action.
	 * @return false||object
	 */
	private function api_request( $_action, $_data ) {

		global $wp_version;

		$data = array_merge( $this->api_data, $_data );

		if( $data['slug'] != $this->slug )
			return;

		if( empty( $data['license'] ) )
			return;

		$api_params = array(
			'edd_action' 	=> 'get_version',
			'license' 		=> $data['license'],
			'name' 			=> $data['item_name'],
			'slug' 			=> $this->slug,
			'author'		=> $data['author']
		);
		$request = wp_remote_post( $this->api_url, array( 'timeout' => 15, 'sslverify' => false, 'body' => $api_params ) );

		if ( ! is_wp_error( $request ) ):
			$request = json_decode( wp_remote_retrieve_body( $request ) );
			if( $request && isset( $request->sections ) )
				$request->sections = maybe_unserialize( $request->sections );
			return $request;
		else:
			return false;
		endif;
	}
}

bean-pricingtables.php
<?php
/**
 * Plugin Name: Bean Pricing Tables
 * Plugin URI: http://themebeans.com/plugin/bean-pricing-tables/?ref=plugin_bean_pricing
 * Description: Enables shortcode pricing tables to be added to your theme.
 * Version: 1.3
 * Author: ThemeBeans
 * Author URI: http://themebeans.com/?ref=plugin_bean_pricing
 *
 *
 * @package Bean Plugins
 * @subpackage PricingTables
 * @author ThemeBeans
 * @since PricingTables 1.0
 */




/*===================================================================*/
/*	MAKE SURE WE DO NOT EXPOSE ANY INFO IF CALLED DIRECTLY
/*===================================================================*/
if ( !function_exists( 'add_action' ) )
{
	echo 'Hi there!  I\'m just a plugin, not much I can do when called directly.';
	exit;
}


/*===================================================================*/
/*
/* PLUGIN FEATURES SETUP
/*
/*===================================================================*/

$bean_plugin_features[ plugin_basename( __FILE__ ) ] = array(
        "updates"  => false // Whether to utilize plugin updates feature or not
    );


if ( ! function_exists( 'bean_plugin_supports' ) ) 
{
    function bean_plugin_supports( $plugin_basename, $feature ) 
    {
        global $bean_plugin_features;

        $setup = $bean_plugin_features;

        if( isset( $setup[$plugin_basename][$feature] ) && $setup[$plugin_basename][$feature] )
            return true;
        else
            return false;
    }
}


/*===================================================================*/
/*
/* PLUGIN UPDATER FUNCTIONALITY
/*
/*===================================================================*/
define( 'EDD_BEANPRICINGTABLES_TB_URL', 'http://themebeans.com' );
define( 'EDD_BEANPRICINGTABLES_NAME', 'Bean Pricing Tables' );

if ( bean_plugin_supports ( plugin_basename( __FILE__ ), 'updates' ) ) : // check to see if updates are allowed; only import if so

//LOAD UPDATER CLASS
if( !class_exists( 'EDD_SL_Plugin_Updater' ) ) 
{
	include( dirname( __FILE__ ) . '/updates/EDD_SL_Plugin_Updater.php' );
}
//INCLUDE UPDATER SETUP
include( dirname( __FILE__ ) . '/updates/EDD_SL_Activation.php' );


endif; // END if ( bean_plugin_supports ( plugin_basename( __FILE__ ), 'updates' ) )


/*===================================================================*/
/* UPDATER SETUP
/*===================================================================*/
function beanpricingtables_license_setup() 
{
	add_option( 'edd_beanpricingtables_activate_license', 'BEANPRICINGTABLES' );
	add_option( 'edd_beanpricingtables_license_status' );
}
add_action( 'init', 'beanpricingtables_license_setup' );

function edd_beanpricingtables_plugin_updater() 
{
    // check to see if updates are allowed; don't do anything if not
    if ( ! bean_plugin_supports ( plugin_basename( __FILE__ ), 'updates' ) ) return;

	//RETRIEVE LICENSE KEY
	$license_key = trim( get_option( 'edd_beanpricingtables_activate_license' ) );

	$edd_updater = new EDD_SL_Plugin_Updater( EDD_BEANPRICINGTABLES_TB_URL, __FILE__, array( 
			'version' => '1.3',
			'license' => $license_key,
			'item_name' => EDD_BEANPRICINGTABLES_NAME,
			'author' 	=> 'Rich Tabor / ThemeBeans'
		)
	);
}
add_action( 'admin_init', 'edd_beanpricingtables_plugin_updater' );


/*===================================================================*/
/* DEACTIVATION HOOK - REMOVE OPTION
/*===================================================================*/
function beanpricingtables_deactivate() 
{
	delete_option( 'edd_beanpricingtables_activate_license' );
	delete_option( 'edd_beanpricingtables_license_status' );
}
register_deactivation_hook( __FILE__, 'beanpricingtables_deactivate' );








/*===================================================================*/
/*
/* BEGIN BEAN PRICING TABLES PLUGIN
/*
/*===================================================================*/
/*===================================================================*/
/*	PLUGIN CLASS
/*===================================================================*/
if ( ! class_exists( 'Bean_BeanPricingTables' ) ) :
	class Bean_BeanPricingTables {

	    private $BEAN_TINYMCE_URI;
	    private $BEAN_TINYMCE_DIR;




		/*===================================================================*/
		/*	CONSTRUCT
		/*===================================================================*/
	    function __construct()
	    {
	    	require_once( DIRNAME(__FILE__) . '/bean-theme-pricingtables.php' );

	        $this->BEAN_TINYMCE_URI = plugin_dir_url(__FILE__) .'tinymce';
	        $this->BEAN_TINYMCE_DIR = DIRNAME(__FILE__) .'/tinymce';

	        add_action('init', array(&$this, 'action_admin_init'));
	        add_action('admin_enqueue_scripts', array(&$this, 'action_admin_scripts_init'));
	        add_filter('the_posts', array(&$this, 'add_frontend_scripts_if_shortcode_being_used'));
		}




		/*===================================================================*/
		/*	ENQUEUE FRONTEND
		/*===================================================================*/
		function add_frontend_scripts_if_shortcode_being_used($posts)
		{
            if (empty($posts)) return $posts;

            $shortcode_found = false;
            foreach ($posts as $post) {
                if (stripos($post->post_content, '[pricing_table') !== false) {
                    $shortcode_found = true;
                    break;
                }
            }

            if ($shortcode_found) {
                $default_css_url = plugin_dir_url(__FILE__) . 'css/bean-pricingtables.css';
                $default_js_url = plugin_dir_url(__FILE__) . 'js/bean-pricingtables.js';

                wp_enqueue_style( 'bean-pricingtables-style', $default_css_url, false, '1.0', 'all' );
                wp_enqueue_script('bean-pricingtables', $default_js_url, 'jquery', '1.0', true);
            }

            return $posts;
		}




		/*===================================================================*/
		/*	ENQUEUE ADMIN
		/*===================================================================*/
		function action_admin_scripts_init()
		{
			wp_enqueue_style( 'bean-pricingtable-popup', $this->BEAN_TINYMCE_URI . '/css/popup.css', false, '1.0', 'all' );
			wp_localize_script( 'jquery', 'BeanPricingTables', array('plugin_folder' => plugin_dir_url(__FILE__)) );
		}




		/*===================================================================*/
		/*	REGISTERS TINYMCE RICH EDITOR BUTTONS
		/*===================================================================*/
		function action_admin_init()
		{
			if ( ! current_user_can('edit_posts') && ! current_user_can('edit_pages') )
				return;

			if ( get_user_option('rich_editing') == 'true' && is_admin() )
			{
				add_filter( 'mce_external_plugins', array(&$this, 'add_rich_plugins') );
				add_filter( 'mce_buttons', array(&$this, 'register_rich_buttons') );
			}
		}




		/*===================================================================*/
		/*	DEFINE TINYMCE RICH EDITOR JS PLUGIN
		/*===================================================================*/
		function add_rich_plugins( $plugin_array )
		{
            if ( version_compare( get_bloginfo('version'), '3.9', '>=' ) ) {
                $plugin_array['BeanPricingTables'] = $this->BEAN_TINYMCE_URI . '/plugin.js';
            } else {
                $plugin_array['BeanPricingTables'] = $this->BEAN_TINYMCE_URI . '/plugin-pre3.9.js';
            }

			return $plugin_array;
		}




		/*===================================================================*/
		/*	ADDS TINYMCE BUTTON
		/*===================================================================*/
		function register_rich_buttons( $buttons )
		{
			array_push( $buttons, "|", 'bean_pricingtable_button' );
			return $buttons;
		}
	}

	new Bean_BeanPricingTables;

endif; //END class Bean_BeanPricingTables